operator/v1/types_cloudcredential.go

```diff
@@ -0,0 +1,65 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudCredential provides a means to configure an operator to manage CredentialsRequests.
+type CloudCredential struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec CloudCredentialSpec `json:"spec,omitempty"`
+	// +optional
+	Status CloudCredentialStatus `json:"status,omitempty"`
+}
+
+// CloudCredentialsMode is the specified mode the cloud-credential-operator
+// should reconcile CredentialsRequest with
+// +kubebuilder:validation:Enum="";manual;mint;passthrough
+type CloudCredentialsMode string
+
+const (
+	// CloudCredentialsModeManual tells cloud-credential-operator to not reconcile any CredentialsRequests
+	// (primarily used for the disconnected VPC use-cases).
+	CloudCredentialsModeManual CloudCredentialsMode = "manual"
+
+	// CloudCredentialsModeMint tells cloud-credential-operator to reconcile all CredentialsRequests
+	// by minting new users/credentials.
+	CloudCredentialsModeMint CloudCredentialsMode = "mint"
+
+	// CloudCredentialsModePassthrough tells cloud-credential-operator to reconcile all CredentialsRequests
+	// by copying the cloud-specific secret data.
+	CloudCredentialsModePassthrough CloudCredentialsMode = "passthrough"
+)
+
+// CloudCredentialSpec is the specification of the desired behavior of the cloud-credential-operator.
+type CloudCredentialSpec struct {
+	OperatorSpec `json:",inline"`
+	// CredentialsMode allows informing CCO that it should not attempt to dynamically
+	// determine the root cloud credentials capabilities, and it should just run in
+	// the specified mode.
+	// It also allows putting the operator into "manual" mode if desired.
+	// +optional
```

Comment:
define what empty means.

Reply:
Added text to explain that the empty string for this field will put things in the mode where CCO will dynamically try to figure things out based on the capabilities of the cluster's cloud creds (for the clouds where this is supported).

Reply:
Document all constants here please. They won't show up in `kubectl explain` or web docs automatically.